src/server/pps/cmds/cmds.go

```diff
@@ -49,6 +49,56 @@ func (arr ByCreationTime) Less(i, j int) bool {
 	return false
 }
 
+// configReader helps with unmarshalling pipeline configs from JSON. It's used by
+// create-pipeline and update-pipeline
+type configReader struct {
```

Comment:
I think this would be better named `pipelineReader`, config seems too generic to me.